lib/generateClass.js

```diff
@@ -301,29 +301,42 @@ function writeToFhir(def, fhir, cw) {
 
    /* UNCOMMENT BELOW CODE TO JAM IN DUMMY MAPPING DATA */ 
 
-    // if(fhirProfile.type === 'Patient'){
-    //   const addMapping = (path, map) => { fhirProfile.snapshot.element.find(f => f.path === path).mapping.push(map)};
-    //   addMapping('Patient.active', {'identity': 'shr', 'map': '<shr.fhir.ActiveFlag>'});
-    //   addMapping('Patient.birthDate', {'identity': 'shr', 'map': '<Value>'});
+    // const addMapping = (path, map) => {
+    //   const el = fhirProfile.snapshot.element.find(f => f.path === path);
+    //   el.mapping = el.mapping.filter(m => m.identity !== 'shr');
+    //   el.mapping.push(map);
+    // };
+    // if(fhirProfile.name === 'PatientEntryProfile'){
+    //   // addMapping('Patient.active', {'identity': 'shr', 'map': '<shr.simple.BooleanValue>'}); // DONE
+    //   // addMapping('Patient.birthDate', {'identity': 'shr', 'map': '<Value>'}); // DONE
+    //   addMapping('Patient.name.text', {'identity': 'shr', 'map': '<shr.simple.StringValue>'});
+    // }
+    // if(fhirProfile.name === 'PractitionerEntryProfile'){
+    //   addMapping('Practitioner.active', {'identity': 'shr', 'map': '<shr.simple.DoubleNestedBooleanValue>.<shr.simple.NestedBooleanValue>.<shr.simple.BooleanValue>'});
+    //   addMapping('Practitioner.name.text', {'identity': 'shr', 'map': '<shr.simple.NestedStringValue>.<shr.simple.StringValue>'});
     // }
 
     for (element of fhirProfile.snapshot.element) {
       const mapping = element.mapping && element.mapping.find(m => m['identity'] === 'shr');
 
       if(mapping !== undefined){
-        const match = mapping.map.match(/\<([^\>]*)>/);
+        // const match = mapping.map.match(/\<([^\>]*)>/);
+        const match = mapping.map.match(/\<([^\>]*)>/g)
         if(match){
-          const name = match[1].split('.').pop();
-          const fhirName = element.path.split('.')[1]; // Currently only handling most basic case
-          if(name === 'Value'){
+          const name = match.map(e => toSymbol(e.slice(1).split('.').pop().slice(0,-1))).join('.')
+          // const name = match[1].split('.').pop();
+          let fhirName = element.path.split('.');
+          fhirName.shift();
+          fhirName = fhirName.join('.');
+          if(name === 'value'){
             if (def.value instanceof IdentifiableValue && def.value.identifier.isPrimitive) {
-              generateAssignmentIfListFHIR(def.value.card, fhirName, toSymbol(name), cw);
+              generateAssignmentIfListFHIR(def.value.card, fhirName, name, cw);
             } else {
               console.error('ERROR: Value referenced in mapping but none exist on this element.');
             }
           } else {
-            const field = def.fields.find(f => f.identifier.name === name);
-            generateAssignmentIfListFHIR(field.card, fhirName, toSymbol(name), cw);
+            const field = def.fields.find(f => f.identifier.name === match[0].split('.').slice(-1)[0].slice(0,-1));
```

Comment:
Might it be worth moving this bit of code out into its own function? It's used in two places in this snippet, and is non-intuitive if you don't know what you're looking at. Moving it to a function that can be appropriately named/commented might make it a little more clear.